app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -760,6 +760,62 @@ class LuciBuildService {
     return <Tuple<Target, Task, int>>[];
   }
 
+  /// Schedules [targets] for building of prod artifacts while in a merge queue.
+  Future<void> scheduleMergeGroupBuilds({
+    required Commit commit,
+    required List<Target> targets,
+  }) async {
+    final buildRequests = <bbv2.BatchRequest_Request>[];
+
+    Set<String> availableBuilderSet;
+    try {
+      availableBuilderSet = await getAvailableBuilderSet(
+        project: 'flutter',
+        bucket: 'prod',
+      );
+    } catch (error) {
+      throw 'Failed to get buildbucket builder list due to $error';
+    }
+    log.info('Available builder list: $availableBuilderSet');
+    for (var target in targets) {
+      // Non-existing builder target will be skipped from scheduling.
```

Comment:
If this happens; initializeCiStagingDocument() will have added targets that will not complete and block the queue.

Post submits do something similar, so I'm wondering how often this actually occurs.

Reply:
I solved this in the caller to this new method by fetching the available builds and filtering there. I need to know ahead of time the valid targets so I can write the initial staging document. That should be written before we schedule something.  It would be nicer to have:

for (actual targets) {
    create check run
}
initialize staging
for (actual targets) {
    schedule
}

but that is techdebt for the future and would be less of a problem if we only had one checkrun per pr/push/merge (i.e. Cocoon tracks all tests)